src/app/playground/page.tsx

```diff
@@ -0,0 +1,87 @@
+"use client"
+
+import { useEffect, useState } from "react";
+import init, { take_snapshot } from "codesnap.wasm";
+
+export default function Playground() {
+    const [loaded, setLoaded] = useState(false)
+    const [code, setCode] = useState('print "Hello, World!"')
+    const [lang, setLang] = useState('python')
+    const [imgUrl, setImgUrl] = useState('')
+
+    // TODO: Fetch supported languages from the wasm module, need to consider "before init" behavior
+    const supportedLanguages = [
+        {code: 'python', name: 'Python'},
+        {code: 'rust', name: 'Rust'},
+        {code: 'javascript', name: 'JavaScript'},
+        {code: 'typescript', name: 'TypeScript'},
+        {code: 'java', name: 'Java'},
+        {code: 'c', name: 'C'},
+        {code: 'cpp', name: 'C++'},
+        {code: 'csharp', name: 'C#'},
+        {code: 'go', name: 'Go'},
+        {code: 'ruby', name: 'Ruby'},
+        {code: 'php', name: 'PHP'},
+    ];
+
+    init().then(() => { setLoaded(true) })
```

Comment:
This has to laod a 11.2MB file currently, maybe there could be improvements to the wasm package?